src/runtime/tests/manifest-parser-test.ts

```diff
@@ -866,6 +866,46 @@ describe('manifest parser', () => {
         bar: writes [Bar {y: Number}] = from p in foo.x where p < 10 select new Bar {y: foo.x}
       `);
     });
+    it('parses from/select expression with let with constant', () => {
+      parse(`
+      particle Converter
+        foo: reads [Foo {x: Number}]
+        bar: writes [Bar {y: Number}] =
+          from f in foo
+          let y = 10
+          select new Bar {y: y}
+      `);
+    });
+    it('parses from/select expression with let with expression', () => {
+      parse(`
+      particle Converter
+        foo: reads [Foo {x: Number, y: Number}]
+        bar: writes [Bar {y: Number}] =
+          from f in foo
+          let y = f.x * f.y + 42
+          select new Bar {y: y}
+      `);
+    });
+    it('parses from/select expression with let with function call', () => {
+      parse(`
+      particle Converter
+        foo: reads [Foo {x: List<Number>}]
+        bar: writes [Bar {y: Number}] =
+          from f in foo
+          let y = first(from x in f.x select x / 2)
+          select new Bar {y: y}
+      `);
+    });
+    it('parses from/select expression with let with paxel expression', () => {
+      parse(`
+      particle Converter
+        foo: reads [Foo {x: List<Number>}]
+        bar: writes [Bar {y: Number}] =
+          from f in foo
+          let y = (from x in f.x where x > 42 select x / 2)
```

Comment:
ahh, this is why you added the double-select check, because this probably originally parsed as

```
let y = from x in f.x where x > 42 select x / 2 select new Bar { y: first(y) }
```

When you forgot to put the parenthesis , right?

Reply:
yeah, I played around with it and it was parsing as you say, so adding the check for double select at least hints you at the problem (I hope) :)